src/libcore/marker.rs

```diff
@@ -339,7 +339,18 @@ pub trait Copy : Clone {
 /// [transmute]: ../../std/mem/fn.transmute.html
 #[stable(feature = "rust1", since = "1.0.0")]
 #[lang = "sync"]
-#[rustc_on_unimplemented = "`{Self}` cannot be shared between threads safely"]
+#[rustc_on_unimplemented(
+    on(
+        _Self="std::sync::mpsc::Receiver<T>",
+        label="`{Self}` cannot be shared safely, consider marking the closure `move`"
```

Comment:
I forget if I said this before -- but we don't *know* that a closure is involved, do we? It seems odd to write "the closure" here.

Reply:
We don't, but it does seem to be the most common case that happens in the wild.